datafusion/expr-common/src/type_coercion/aggregates.rs

```diff
@@ -144,260 +106,3 @@ pub fn check_arg_count(
     }
     Ok(())
 }
-
-/// Function return type of a sum
-pub fn sum_return_type(arg_type: &DataType) -> Result<DataType> {
```

Comment:
All of these functions except the avg ones were unused in our code, and I don't think it makes sense to have them available for users anyway